### PR TITLE
added cs application variables to myriad

### PR DESCRIPTION
**Pull request by ashleyschuett**
@nicktate @normanjoyner 

save cs application variables in myriad

### CODE REVIEW

**Comment by ashleyschuett** [comment]
https://github.com/containership/containership.core.constants/pull/3 needs to be merged and released first

File: lib/applications.js [hunk 1]
```diff
@@ -70,44 +71,61 @@ class Applications {
...
+                                    deployed_containers.push(container);
+                                    return callback();
+                                });
+                            }, function (err) {
+                                if (err) {
+                                    return add_callback(err);
+                                }
+
+                                config.containers = deployed_containers;
+                                config.containers = deployed_containers;
```
Comment (normanjoyner) [inline lib/applications.js]: line should be blank

File: lib/applications.js [hunk 2]
```diff
@@ -70,44 +71,61 @@ class Applications {
...
-                        async.each(containers, (container, callback) => {
-                            if(_.has(container.tags, 'constraints') && _.has(container.tags.constraints, 'per_host') && _.has(container.tags, 'host')) {
-                                delete container.tags.host;
-                            }
+                        const auto_variables = {};
 
-                            if(container.random_host_port) {
-                                container.host_port = null;
-                            }
+                        const application_discover_port = ['CS', 'DISCOVERY', 'PORT', application_name.toUpperCase()].join('_');
```
Comment (normanjoyner) [inline lib/applications.js]: `application_discovery_port` just for consistency

File: lib/applications.js [hunk 3]
```diff
@@ -70,44 +71,61 @@ class Applications {
...
+                        auto_variables[application_discover_port] = config.discovery_port;
 
-                            container.status = 'unloaded';
-                            container.host = null;
-                            container.start_time = null;
+                        const applicaton_address = ['CS', 'ADDRESS', application_name.toUpperCase()].join('_');
```
Comment (normanjoyner) [inline lib/applications.js]: `application_address`

File: lib/applications.js [hunk 4]
```diff
@@ -70,44 +71,61 @@ class Applications {
...
+                        auto_variables[applicaton_address] = [application_name, this.core.cluster_id, 'containership'].join('.');
 
-                            self.deploy_container(config.id, container, () => {
-                                deployed_containers.push(container);
-                                return callback();
-                            });
-                        }, function (err) {
-                            if (err) {
+                        return async.eachOf(auto_variables, (value, variable_name, callback) => {
```
Comment (normanjoyner) [inline lib/applications.js]: how is a written auto variable cleaned up if it succeeds but is succeeded by a failure?

File: lib/applications.js [hunk 5]
```diff
@@ -70,44 +71,61 @@ class Applications {
...
+                                if (err) {
+                                    return add_callback(err);
+                                }
+
+                                config.containers = deployed_containers;
+                                config.containers = deployed_containers;
+                                return self.core.scheduler.leader.application.start_health_check(config.id, (err) => {
+                                    // if health check is unable to be started remove application
+                                    if(err) {
+                                        return self.remove(config.id, () => {
```
Comment (normanjoyner) [inline lib/applications.js]: auto variables need cleaned up on removal

File: lib/applications.js [hunk 6]
```diff
@@ -38,12 +38,38 @@ class Applications {
         });
     }
 
+    get_discovery_port_key(application_name) {
```
Comment (nicktate) [inline lib/applications.js]: `get_discovery_port_key_variable`

File: lib/applications.js [hunk 7]
```diff
@@ -38,12 +38,38 @@ class Applications {
...
+        const application_discovery_port = ['CS', 'DISCOVERY', 'PORT', application_name.toUpperCase()].join('_');
+        return application_discovery_port;
+    }
+
+    get_application_address(application_name) {
```
Comment (nicktate) [inline lib/applications.js]: `get_application_address_variable`

File: lib/applications.js [hunk 8]
```diff
@@ -70,44 +96,65 @@ class Applications {
...
+                                if (err) {
+                                    return add_callback(err);
+                                }
+
+                                config.containers = deployed_containers;
+
+                                return self.core.scheduler.leader.application.start_health_check(config.id, (err) => {
+                                    // if health check is unable to be started remove application
+                                    if(err) {
+                                        return self.remove(config.id, () => {
```
Comment (nicktate) [inline lib/applications.js]: application_name

File: lib/applications.js [hunk 9]
```diff
@@ -70,44 +96,65 @@ class Applications {
...
-                                return callback();
-                            });
-                        }, function (err) {
-                            if (err) {
-                                return add_callback(err);
+                        return async.eachOf(auto_variables, (value, variable_name, callback) => {
+                            self.core.cluster.myriad.persistence.set([constants.myriad.AUTOGENERATED_VARIABLES_PREFIX, variable_name].join(constants.myriad.DELIMITER), value, callback);
+                        }, (err) => {
+                            if(err) {
+                                return self.cleanup_autogenerated_variable(application_name, () => {
```
Comment (nicktate) [inline lib/applications.js]: wrap this in `return self.remove(config.id, () => {`, if we fail to set the autogenerated variables we should just delete teh app

File: lib/applications.js [hunk 10]
```diff
@@ -70,44 +96,65 @@ class Applications {
...
-                                return add_callback(null, config);
+                                if(container.random_host_port) {
+                                    container.host_port = null;
+                                }
+
+                                container.status = 'unloaded';
+                                container.host = null;
+                                container.start_time = null;
+
+                                self.deploy_container(config.id, container, () => {
```
Comment (nicktate) [inline lib/applications.js]: application_name

File: lib/applications.js [hunk 11]
```diff
@@ -70,44 +96,65 @@ class Applications {
...
+                                    return callback();
+                                });
+                            }, function (err) {
+                                if (err) {
+                                    return add_callback(err);
+                                }
+
+                                config.containers = deployed_containers;
+
+                                return self.core.scheduler.leader.application.start_health_check(config.id, (err) => {
```
Comment (nicktate) [inline lib/applications.js]: application_name

**Pull request merged**